src/cpp/server/server_context.cc

```diff
@@ -182,7 +185,14 @@ void ServerContext::TryCancel() const {
 }
 
 bool ServerContext::IsCancelled() const {
```

Comment:
Question: Previously, if someone called AsyncNotifyWhenDone() [ and lets say the tag was delivered as well] and then called IsCancelled(), then TryPluck() was called again. Was this a bug ? (Just making sure that the reason for your refactoring here is to avoid calling TryPluck() for this reason)